pkg/shp/flags/core_envvar_value.go

```diff
@@ -0,0 +1,61 @@
+package flags
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// CoreEnvVarSliceValue implements pflag.Value interface, in order to store corev1.EnvVar key-value
+// pairs, used on Shipwright's BuildSpec.
+type CoreEnvVarSliceValue struct {
+	envs *[]corev1.EnvVar // pointer to the slice of EnvVar
+}
+
+// String prints out the string representation of the slice of EnvVar objects.
+func (c *CoreEnvVarSliceValue) String() string {
+	s := []string{}
+	for _, e := range *c.envs {
+		s = append(s, fmt.Sprintf("%s=%s", e.Name, e.Value))
+	}
+	return fmt.Sprintf("[%s]", s)
+}
+
+// setEnvVar populates the local slice of EnvVar, making sure the entry is not repeated.
+func (c *CoreEnvVarSliceValue) setEnvVar(value string) error {
+	k, v, err := splitKeyValue(value)
+	if err != nil {
+		return err
+	}
+	for _, e := range *c.envs {
+		if k == e.Name {
+			return fmt.Errorf("environment variable '%s' is already set!", k)
+		}
+	}
+	*c.envs = append(*c.envs, corev1.EnvVar{Name: k, Value: v})
+	return nil
+}
+
+// Set receives a string with key-value pairs comma separated.
+func (c *CoreEnvVarSliceValue) Set(value string) error {
+	values, err := readAsCSV(value)
```

Comment:
Thanks @SaschaSchwarze0, indeed we need to handle each entry without having to parse CSV. I've changed the implementation to use `stringArray` instead of `stringSlice`, so we have have the [desired behavior](https://github.com/spf13/pflag/issues/157#issuecomment-366286043) you explained, please consider.